plugin/src/main/cpp/extensions/openxr_fb_composition_layer_image_layout_extension_wrapper.cpp

```diff
@@ -0,0 +1,127 @@
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "extensions/openxr_fb_composition_layer_image_layout_extension_wrapper.h"
+
+#include <godot_cpp/classes/open_xrapi_extension.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+using namespace godot;
+
+static const char *VERTICAL_FLIP_PROPERTY_NAME = "XR_FB_composition_layer_image_layout/vertical_flip";
+
+OpenXRFbCompositionLayerImageLayoutExtensionWrapper *OpenXRFbCompositionLayerImageLayoutExtensionWrapper::singleton = nullptr;
+
+OpenXRFbCompositionLayerImageLayoutExtensionWrapper *OpenXRFbCompositionLayerImageLayoutExtensionWrapper::get_singleton() {
+	if (singleton == nullptr) {
+		singleton = memnew(OpenXRFbCompositionLayerImageLayoutExtensionWrapper());
+	}
+	return singleton;
+}
+
+OpenXRFbCompositionLayerImageLayoutExtensionWrapper::OpenXRFbCompositionLayerImageLayoutExtensionWrapper() :
+		OpenXRExtensionWrapperExtension() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbCompositionLayerImageLayoutExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_COMPOSITION_LAYER_IMAGE_LAYOUT_EXTENSION_NAME] = &fb_composition_layer_image_layout;
+	singleton = this;
+}
+
+OpenXRFbCompositionLayerImageLayoutExtensionWrapper::~OpenXRFbCompositionLayerImageLayoutExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbCompositionLayerImageLayoutExtensionWrapper::_bind_methods() {
+}
+
+void OpenXRFbCompositionLayerImageLayoutExtensionWrapper::cleanup() {
+	fb_composition_layer_image_layout = false;
+}
+
+Dictionary OpenXRFbCompositionLayerImageLayoutExtensionWrapper::_get_requested_extensions() {
+	Dictionary result;
+	for (auto ext : request_extensions) {
+		uint64_t value = reinterpret_cast<uint64_t>(ext.value);
+		result[ext.key] = (Variant)value;
+	}
+	return result;
+}
+
+uint64_t OpenXRFbCompositionLayerImageLayoutExtensionWrapper::_set_viewport_composition_layer_and_get_next_pointer(const void *p_layer, const Dictionary &p_property_values, void *p_next_pointer) {
+	if (!fb_composition_layer_image_layout) {
+		return reinterpret_cast<uint64_t>(p_next_pointer);
+	}
+
+	const XrCompositionLayerBaseHeader *layer = reinterpret_cast<const XrCompositionLayerBaseHeader *>(p_layer);
+
+	if (!layer_structs.has(layer)) {
+		layer_structs[layer] = {
+			XR_TYPE_COMPOSITION_LAYER_IMAGE_LAYOUT_FB, // type
+			nullptr, // next
+		};
```

Comment:
I know this is being set at the bottom of this function, but I'd had it in my head for some reason that when initializing like this, we should still explicitly write out all fields when possible.

```suggestion
		layer_structs[layer] = {
			XR_TYPE_COMPOSITION_LAYER_IMAGE_LAYOUT_FB, // type
			nullptr, // next
			0 // flags
		};
```

Reply:
Fair enough!